presto-main/src/main/java/com/facebook/presto/operator/scalar/sql/ArrayArithmeticFunctions.java

```diff
@@ -78,4 +78,52 @@ public static String arrayFrequencyVarchar()
                 "(m, x) -> IF (x IS NOT NULL, MAP_CONCAT(m,MAP_FROM_ENTRIES(ARRAY[ROW(x, COALESCE(ELEMENT_AT(m,x) + 1, 1))])), m)," +
                 "m -> m)";
     }
+
+    @SqlInvokedScalarFunction(value = "array_dupes", deterministic = true, calledOnNullInput = false)
+    @Description("Returns set of elements that have duplicates")
+    @SqlParameter(name = "input", type = "array(varchar)")
+    @SqlType("array(varchar)")
+    public static String arrayDupesVarchar()
+    {
+        return "RETURN CONCAT(" +
+                "CAST(IF (cardinality(filter(input, x -> x is NULL)) > 1, ARRAY[NULL], ARRAY[]) AS ARRAY(VARCHAR))," +
+                "reduce(" +
+                "input," +
+                "MAP()," +
+                "(m, x) -> IF (x IS NOT NULL, MAP_CONCAT(m, MAP_FROM_ENTRIES(ARRAY[ROW(x, COALESCE(ELEMENT_AT(m,x) + 1, 1))])), m)," +
+                "m -> map_keys(map_filter(m, (k, v) -> v > 1))))";
```

Comment:
I think we already have array_frequency. If so just use that and do something like:

map_keys(map_filter(array_frequency(input), (k, v) -> v > 1))

or even better performant one:

reduce(array_sort(input), array[], (s, x) -> if(element_at(s, -1) = x, s, s || x))

Reply:
Please double check, we might not be allowing using sql functions inside sql functions. We probably don't need to be this strict. It should be more like not allowing dynamic functions in other dynamic functions (I don't remember what we were actually checking)